### PR TITLE
Swap BeaconState to v0.2.0 shuffling algorithm

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

NA

## Proposed Changes

Implements the new shuffling algorithm for `BeaconState`

## Additional Info

NA

### CODE REVIEW

File: eth2/types/src/beacon_state_tests.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
+use crate::{
```
Comment (AgeManning) [inline eth2/types/src/beacon_state_tests.rs]: Should we put this under a  #[cfg(test)] ? Or are we using the BeaconStateTestBuilder elsewhere?

Comment (AgeManning) [inline eth2/types/src/beacon_state_tests.rs]: What do you think about putting the tests within their own directory, and importing from the local file. So we don't need to load from lib.rs
More specifically, moving beacon_state_tests.rs to ./beacon_state/tests.rs and adding `mod tests` to `beacon_state.rs`

Comment (paulhauner) [reply eth2/types/src/beacon_state_tests.rs]: Sounds good. Do you think `beacon_state.rs` should live in `beacon_state/` or just the tests?

Comment (AgeManning) [reply eth2/types/src/beacon_state_tests.rs]: Pierre does it with just the tests, so would be leaning towards that approach.

Comment (paulhauner) [reply eth2/types/src/beacon_state_tests.rs]: Cool, works for me. I'll do it now.

Comment (paulhauner) [reply eth2/types/src/beacon_state_tests.rs]: Done!

**Pull request merged**